server/service/src/sync/translations/store.rs

```diff
@@ -26,6 +30,9 @@ pub struct LegacyStoreRow {
     #[serde(deserialize_with = "empty_str_as_option_string")]
     logo: Option<String>,
     store_mode: LegacyStoreMode,
+    #[serde(deserialize_with = "zero_date_as_option")]
+    #[serde(serialize_with = "date_option_to_isostring")]
+    pub created_date: Option<NaiveDate>,
```

Comment:
How does this work? I don't have a created_date column in mSupply is it stored somewhere else in 4D?

Reply:
<img width="486" alt="Screenshot 2024-03-08 at 07 45 08" src="https://github.com/msupply-foundation/open-msupply/assets/61820074/a1a2462b-13e4-4d66-a204-d86dda66459c">
It is stored in the store table and indicates when the store was created

Reply:
thanks seeing it now. This what I was seeing before in the 4D UI: 
![image](https://github.com/msupply-foundation/open-msupply/assets/88299195/33044992-7a7e-4979-b351-3aa4de7b2c9a)
But if you scroll down there is another item! :)

So in 4D its really a nullable value. Wonder if we should make it non-nullable in omSupply, i.e. map NULL to the date at `0`.

We should also already use a `create_datetime` in omSupply. Don't see a reason why we should not also record the time value in the future (when creating a store in omSupply central server) and then we already have the correct type in omSupply. @Chris-Petty I guess it only stores a date because there is not datetime DB type in 4D?

Reply:
Normally with those there's a created_time field in the table as well on 4d... Is time really important when creating the store? 🤔  Can add it in ig, but usually on 4d there's a created time field too 🤔

Reply:
time would be useful for debugging but I guess there is also a log entry for it?

Reply:
Yea I think you both have it - 4D doesn't have date/time, and given time doesn't matter so much we didn't bother adding a field for it. So I think `Option(NaiveDate)` seems as good as it gets - would love for it to not be Option though...

It's not really `Null` in mSupply, I'm pretty sure it's the "blank" date value , literally `0000-00-00`... make of that what you will lol. Might as well be Null. 
![image](https://github.com/msupply-foundation/open-msupply/assets/7684221/98e83893-ce90-4887-a125-bb1acaa1aa2e)
(The last checkbox will do this, it's on for the majority of fields in mSupply except `om_` fields I think.)

That said, when this was added there was a migration code that tried to guess the created date based on logs or invoices:
https://github.com/msupply-foundation/msupply/blob/157aad17c99775ac720276d3e03b7b56b79ebb3d/Project/Sources/Methods/startup_update_version.4dm#L4397-L4418
Admittedly it might still fail and leave the blank value if there is no transacts nor ANY log (could be a datafile from 15 years before logging and the store was was never used??)... maybe failing those two it'd been better to use the earliest `store.created_date` from other stores... could still add a new migration and guarantee it 😆.

@clemens-msupply haha yea the sizing thing hiding fields is super dumb in 4D 😆. You can resize everything by pressing `ctrl+a` to select all tables in the schema viewer, then the cogs at the top has an auto resize. If only it reran every time a field was added... 
![image](https://github.com/msupply-foundation/open-msupply/assets/7684221/d65746a3-6be8-4449-9118-1035b787479c)

Reply:
That said I don't mind using `NaiveDatetime` if it makes the logic checking `transact.created_datetime < (store.created_date(time)- 1month)` easier

Reply:
I think we should at least use `pub created_date: NaiveDate` in open mSupply then right? and map "Null" or "0000-00-00" to the earliest available date? But this would also mean to reintegrate all stores to get the correct created_date...

Reply:
Can you create a carry over issue then if that's what we should do